### ISSUE TITLE
[Proposal] Convolutional Neural Networks

### ISSUE THREAD

**Issue by euler16** [pydatadelhi/talks#43]
- Abstract (2-3 lines)
ConvNets are the default models for Computer Vision tasks relying on a simple linear operation.

- Brief Description and Contents to be covered
Typical ConvNet architecture, what is convolution, math and intuition behind convolution, concept of padding , strides, Deconvolution.

- Pre-requisites for the talk
Understanding about simple MLPs 

- Time required for the talk
45mins

- Link to slides
will upload soon
- Will you be doing hands-on demo as well?
will show construction of ConvNets in Tensorflow and PyTorch. It will be a precooked code setup
- Link to ipython notebook (if any)
will upload it
- About yourself
I am Nilay Shrivastava CS undergrad at NSIT
- Are you comfortable if the talk is recorded and uploaded to PyData Delhi's YouTube channel ?
yes
- Any query ?
I would like this talk to be a part of the series on Machine learning as proposed by @manojpandey   
So ideally this talk should be placed after we have had talks on classical ML algos,linear/logistic regression, basic neural nets. I also propose setting up a group so that we all can coordinate this series

**Comment by rishikksh20** [comment]
@euler16  Actually the thing is most of the people who attend meetups are totally Noob in subject of neural network. So it's better you start with ```Basic Learning representation with Neural network``` then go for CNN or RNN or anything else.  @manojpandey it is very necessary for us to organiser each talk in proper manner and sequence.
So the proper sequence for this series which I supposed would be:
1. **```Basic intuition behind machine learning.```** (Why it is necessary and basic algos and basic data pre-processing and analysis) I think this thing topic already cover but if necessary do it again.
2. **```Optimizing machine learning algos, Boosting, Bagging and Stacking ensemble learning ```** (This topic is very necessary as it help you to win Kaggle competition as well as creating stable model) In real world scenarios all companies prefer ensemble model for stability reason.
3. **```Basic Learning Representation with Neural Network```** (Starting point for Deep learning) Beginning of new world for researchers.
4. **```Understanding of Neural Network (internal representation) and Optimisation Algos```** (How Neural network out perform traditional algos).
5. **```Convolution  Neural network and image recognition problem ```**
6. **```Sequence modelling and forecasting ```**(Basic regression to simple markov models) Starting point for NLP and RNN.
7. **```Natural language Modelling and vector representation```** (Basic natural processing stuff)
8. **```Recurrent Neural network```** (LSTM,GRU)
till this is enough for Machine learning.
For data analysis and Data science concept we can arrange parallel tutorials for **Pandas** , **Scikit-learn**, **Keras** , **Tensorflow** etc and talk related to feature engineering, data cleaning, cross-validation etc.

**Note:** For everybody, feel free to up in and put your suggestion by opening new issue please don't comment it here which might be irrelevant to this issue.

**Comment by euler16** [comment]
I agree thats why I referenced @manojpandey  's suggestion of series. I want to do this after we have had the basics covered.

**Comment by rishikksh20** [comment]
Ok I am creating new issue

**Comment by manojpandey** [comment]
@shagunsodhani - Please have a look ^
Update: (maybe at the new issue then)

**Comment by manojpandey** [comment]
Hey @euler16 
Can you check #26 and see what different would you be talking about ? :))

cc: @ananyahjha93

**Comment by euler16** [comment]
@manojpandey I will be focussing in depth on convolution operation, how backpropagation is actually done through conv layer....different cases of strides and paddings. I will use PyTorch

**Comment by shagunsodhani** [comment]
@manojpandey @euler16 It seems this talk is more about conv and its theory while #45 is more about application of convs for a specific use case. Going by this logic, #43 should be done first and then #45. It would be ideal if both @euler16 and @techedlaksh use the same framework but even then it should be fine.

**Comment by manojpandey** [comment]
Alright ! Thanks @shagunsodhani for the feedback :)
